src/Microsoft.OpenApi.CSharpAnnotations.DocumentGeneration/OpenApiDocumentGenerationSettings.cs

```diff
@@ -16,15 +16,27 @@ public class OpenApiDocumentGenerationSettings
         /// Creates an instance of <see cref="OpenApiDocumentGenerationSettings"/>.
         /// </summary>
         /// <param name="schemaGenerationSettings">The schema generation settings.</param>
-        public OpenApiDocumentGenerationSettings(SchemaGenerationSettings schemaGenerationSettings)
+        /// <param name="removeDuplicateStringFromParamName">Indicates whether to remove duplicate string from parameter
+        /// name to work around rosyln issue. https://github.com/dotnet/roslyn/issues/26292.
+        /// </param>
+        public OpenApiDocumentGenerationSettings(
+            SchemaGenerationSettings schemaGenerationSettings,
+            bool removeDuplicateStringFromParamName = false)
```

Comment:
Can we create a separate constructors instead of using default param?

Reply:
make sense.

---
In reply to: [253639394](https://github.com/Microsoft/OpenAPI.NET.CSharpAnnotations/pull/198#discussion_r253639394) [](ancestors = 253639394)